pycfmodel/model/resources/properties/statement_condition.py

```diff
@@ -0,0 +1,328 @@
+    StringEqualsIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    StringNotEqualsIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    StringEqualsIgnoreCaseIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    StringNotEqualsIgnoreCaseIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    StringLikeIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    StringNotLikeIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    ForAllValuesStringEquals: Optional[Dict[str, ResolvableStrOrList]]
+    ForAllValuesStringNotEquals: Optional[Dict[str, ResolvableStrOrList]]
+    ForAllValuesStringEqualsIgnoreCase: Optional[Dict[str, ResolvableStrOrList]]
+    ForAllValuesStringNotEqualsIgnoreCase: Optional[Dict[str, ResolvableStrOrList]]
+    ForAllValuesStringLike: Optional[Dict[str, ResolvableStrOrList]]
+    ForAllValuesStringNotLike: Optional[Dict[str, ResolvableStrOrList]]
+    ForAllValuesStringEqualsIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    ForAllValuesStringNotEqualsIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    ForAllValuesStringEqualsIgnoreCaseIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    ForAllValuesStringNotEqualsIgnoreCaseIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    ForAllValuesStringLikeIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    ForAllValuesStringNotLikeIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    ForAnyValueStringEquals: Optional[Dict[str, ResolvableStrOrList]]
+    ForAnyValueStringNotEquals: Optional[Dict[str, ResolvableStrOrList]]
+    ForAnyValueStringEqualsIgnoreCase: Optional[Dict[str, ResolvableStrOrList]]
+    ForAnyValueStringNotEqualsIgnoreCase: Optional[Dict[str, ResolvableStrOrList]]
+    ForAnyValueStringLike: Optional[Dict[str, ResolvableStrOrList]]
+    ForAnyValueStringNotLike: Optional[Dict[str, ResolvableStrOrList]]
+    ForAnyValueStringEqualsIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    ForAnyValueStringNotEqualsIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    ForAnyValueStringEqualsIgnoreCaseIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    ForAnyValueStringNotEqualsIgnoreCaseIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    ForAnyValueStringLikeIfExists: Optional[Dict[str, ResolvableStrOrList]]
+    ForAnyValueStringNotLikeIfExists: Optional[Dict[str, ResolvableStrOrList]]
+
+    eval: Optional[Callable]
+
+    @root_validator(pre=True)
+    def remove_colon(cls, values):
+        return {key.replace(":", ""): value for key, value in values.items()}
+
+    @validator(
+        "BinaryEquals",
+        "BinaryEqualsIfExists",
+        "ForAllValuesBinaryEquals",
+        "ForAllValuesBinaryEqualsIfExists",
+        "ForAnyValueBinaryEquals",
+        "ForAnyValueBinaryEqualsIfExists",
+        each_item=True,
+        pre=True,
+    )
+    def validate_binary(cls, value):
+        try:
+            value = b64decode(value)
+        except binascii.Error:
+            raise ValueError(f"Binary value not valid")
```

Comment:
Will this log the value that we failed to decode, or should that be added in the error string. If the message remains unchanged, the `f` at the start can go :)

Reply:
This will be used by pydantic when validatinc an object. So it will flag the error that failed to validate